diff --git a/drivers/power/supply/qcom_pm8150b_charger.c b/drivers/power/supply/qcom_pm8150b_charger.c
--- a/drivers/power/supply/qcom_pm8150b_charger.c
+++ b/drivers/power/supply/qcom_pm8150b_charger.c
@@ -282,7 +282,7 @@ static int smb5_get_prop_usb_online(struct smb5_chip *chip, int *val)
 		return rc;
 	}
 
-	*val = (stat & USBIN_PLUGIN_RT_STS_BIT);
+	*val = !!(stat & USBIN_PLUGIN_RT_STS_BIT);
 	return 0;
 }
 
@@ -310,7 +310,7 @@ static int smb5_apsd_get_charger_type(struct smb5_chip *chip, int *val)
 		return rc;
 	}
 	if (!(apsd_stat & APSD_DTC_STATUS_DONE_BIT)) {
-		dev_err(chip->dev, "Apsd not ready");
+		dev_dbg(chip->dev, "APSD not ready\n");
 		return -EAGAIN;
 	}
 
